src/org/zaproxy/zap/extension/jxbrowser/ZapTabbedPanel.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.jxbrowser;
+
+import java.awt.Component;
+
+import javax.swing.Icon;
+import javax.swing.ImageIcon;
+import javax.swing.JTabbedPane;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+
+import org.zaproxy.zap.utils.DisplayUtils;
+
+
+public class ZapTabbedPanel extends JTabbedPane {
+
+    private static final long serialVersionUID = 1L;
+
+    public ZapTabbedPanel() {
+        
+        this.addChangeListener(new ChangeListener() {
+            @Override
+            public void stateChanged(ChangeEvent e) {
+                setCloseButtonStates();
+            }
+        });
+
+    }
+    
+    public void addTab(String title, Icon icon, final Component c, boolean hideable, boolean visiXbleCanDelete, int index) {
```

Comment:
`visiXbleCanDelete` is a confusing name and seems unused in the method.

`c` same as earlier, could be a more maint friendly name. Assume this applies throughout :grinning: